### PR TITLE
APP-116 + APP-145 pass in menu handlers

### PR DESCRIPTION
**Pull request by tarikstafford**
APP-116 && APP-145

### CODE REVIEW

File: Projects/App/Sources/Generic/AppFlow.swift [hunk 1]
```diff
@@ -48,14 +48,58 @@ struct EmbarkOnboardingFlow: Presentable {
 
         return (viewController, Future { completion in
             bag += storySignal.atValue { story in
+                let embark = Embark(name: story.name, flowType: .onboarding)
+                bag += embark.routeSignal.onValue { route in
```
Comment (sampettersson) [inline Projects/App/Sources/Generic/AppFlow.swift:52]: Maybe its cleaner if embark returns a `FiniteSignal<Either<Route, Redirect>>`?

File: Projects/App/Sources/Generic/ApplicationState.swift [hunk 2]
```diff
@@ -52,7 +52,7 @@ extension ApplicationState {
         case .onboardingChat, .onboarding:
             return window.present(
                 Onboarding(),
-                options: [.defaults],
+                options: [.defaults, .embedInNavigationController],
```
Comment (sampettersson) [inline Projects/App/Sources/Generic/ApplicationState.swift:55]: Isn't `.embedInNavigationController` in `.defaults`? 🤔

File: Projects/App/Sources/Screens/ProfileScreen/Screens/AboutScreen/AppInfo.swift [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+            }
+        }
+
+        func setupAppSettings() {
+            let row = State.InfoRows.locale
+            let market = Localization.Locale.currentLocale.market
+            bag += bodySection.append(AppInfoRow(title: row.title, icon: market.icon, isTappable: row.isTappable, value: market.marketName))
+
+            let language = State.InfoRows.language
+            let languageRow = AppInfoRow(title: language.title, icon: language.icon, isTappable: language.isTappable, value: Localization.Locale.currentLocale.displayName)
```
Comment (sampettersson) [inline Projects/App/Sources/Screens/ProfileScreen/Screens/AboutScreen/AppInfo.swift:172]: This line is rather long, split up maybe?

File: Projects/App/Sources/Screens/ProfileScreen/Screens/AboutScreen/AppInfo.swift [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            bag += bodySection.append(languageRow)
+
+            bag += languageRow.onSelect.onValue { _ in
+                presentAlert()
+            }
+        }
+
+        func setupAppInfo() {
+            [State.InfoRows.memberId, State.InfoRows.version].forEach { row in
+                value(row: row) { rowValue in
```
Comment (sampettersson) [inline Projects/App/Sources/Screens/ProfileScreen/Screens/AboutScreen/AppInfo.swift:182]: Couldn't this lead to the ordering being non-deterministic? 😅

Comment (sampettersson) [reply Projects/App/Sources/Screens/ProfileScreen/Screens/AboutScreen/AppInfo.swift:182]: Maybe we should add the row first and show a spinner until it has a value?

File: Projects/Embark/Sources/Embark.swift [hunk 5]
```diff
@@ -189,44 +194,77 @@ extension Embark: Presentable {
...
             viewController.navigationItem.leftBarButtonItem = backButton
 
-            let tooltipButton = UIBarButtonItem(image: hCoreUIAssets.infoLarge.image, style: .plain, target: nil, action: nil)
+            let tooltipButton = UIButton()
+            tooltipButton.setImage(hCoreUIAssets.infoLarge.image, for: .normal)
+
+            let didTapTooltip = tooltipButton.signal(for: .touchUpInside)
+
+            bag += didTapTooltip
+                .onValue { () in
```
Comment (sampettersson) [inline Projects/Embark/Sources/Embark.swift:203]:
```suggestion
                .onValue {
```

**Review by sampettersson — CHANGES_REQUESTED** [review]
You are acing it!! So clean code!

File: Projects/App/Sources/Generic/AppFlow.swift (hunk 1 above)
Comment (tarikstafford) [reply Projects/App/Sources/Generic/AppFlow.swift:52]: I didn't know this was a thing! CLEAN

File: Projects/App/Sources/Screens/ProfileScreen/Screens/AboutScreen/AppInfo.swift (hunk 4 above)
Comment (tarikstafford) [reply Projects/App/Sources/Screens/ProfileScreen/Screens/AboutScreen/AppInfo.swift]: @sampettersson trueeee

**Review by sampettersson — APPROVED** [review]

**Pull request merged**